src/simularium/OctopusClient.ts

```diff
@@ -1,23 +1,74 @@
 import { TrajectoryType } from "../constants.js";
-import { WebsocketClient, NetMessageEnum } from "./WebsocketClient.js";
+import { FrontEndError } from "./FrontEndError.js";
+import {
+    NetMessageEnum,
+    NetConnectionParams,
+    NetMessage,
+    WebsocketClient,
+} from "./WebsocketClient.js";
 
-export class OctopusServicesClient {
+export class ConversionClient {
     private webSocketClient: WebsocketClient;
-    private lastRequestedFile = "";
+    public onConversionComplete: (fileName: string) => void;
+    public handleError: (error: FrontEndError) => void | (() => void);
+    public lastRequestedFile: string;
 
-    constructor(webSocketClient: WebsocketClient) {
-        this.webSocketClient = webSocketClient;
+    constructor(
+        netConnectionSettings: NetConnectionParams,
+        lastRequestedFile: string,
+        errorHandler?: (error: FrontEndError) => void
+    ) {
+        this.webSocketClient = new WebsocketClient(
+            netConnectionSettings,
+            errorHandler
+        );
+        this.handleError =
+            errorHandler ||
+            (() => {
+                /* do nothing */
+            });
+        this.lastRequestedFile = lastRequestedFile;
+        this.onConversionComplete = () => {
+            /* do nothing */
+        };
+    }
+
+    public disconnect(): void {
+        this.webSocketClient.disconnect();
+    }
+
+    public async initialize(fileName: string): Promise<void> {
+        this.lastRequestedFile = fileName;
+        try {
+            await this.webSocketClient.connectToRemoteServer();
+        } catch (error) {
+            this.handleError(error as FrontEndError);
+        }
     }
```

Comment:
do you really need a separate `initialize` here? If the Conversion client only exists to convert a file on command, and you're already sending in the file to the construction why not just connect in the constructor as well?